src/ralph/backends/lrs/mongo.py

```diff
@@ -0,0 +1,84 @@
+"""MongoDB LRS backend for Ralph."""
+
+import logging
+from typing import List
+
+from bson.objectid import ObjectId
+from pymongo import ASCENDING, DESCENDING
+
+from ralph.backends.data.mongo import MongoDataBackend
+from ralph.backends.lrs.base import (
+    BaseLRSBackend,
+    StatementParameters,
+    StatementQueryResult,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class MongoLRSBackend(BaseLRSBackend, MongoDataBackend):
+    """MongoDB LRS backend."""
+
+    def query_statements(self, params: StatementParameters) -> StatementQueryResult:
+        """Return the results of a statements query using xAPI parameters."""
+        mongo_query_filters = {}
+
+        if params.statementId:
+            mongo_query_filters.update({"_source.id": params.statementId})
+
+        if params.agent:
+            mongo_query_filters.update({"_source.actor.account.name": params.agent})
+
+        if params.verb:
+            mongo_query_filters.update({"_source.verb.id": params.verb})
+
+        if params.activity:
+            mongo_query_filters.update(
+                {
+                    "_source.object.objectType": "Activity",
+                    "_source.object.id": params.activity,
+                },
+            )
+
+        if params.since:
+            mongo_query_filters.update(
+                {"_source.timestamp": {"$gt": params.since.isoformat()}}
+            )
+
+        if params.until:
+            mongo_query_filters.update(
+                {"_source.timestamp": {"$lte": params.until.isoformat()}}
+            )
+
+        if params.search_after:
+            search_order = "$gt" if params.ascending else "$lt"
+            mongo_query_filters.update(
+                {"_id": {search_order: ObjectId(params.search_after)}}
+            )
+
+        mongo_sort_order = ASCENDING if params.ascending else DESCENDING
+        mongo_query_sort = [
+            ("_source.timestamp", mongo_sort_order),
+            ("_id", mongo_sort_order),
+        ]
+
+        query = {
+            "filter": mongo_query_filters,
+            "limit": params.limit,
+            "sort": mongo_query_sort,
+        }
+        mongo_response = list(self.read(query=query, ignore_errors=True))
+        search_after = None
+        if mongo_response:
+            search_after = mongo_response[-1]["_id"]
+
+        return StatementQueryResult(
+            statements=[document["_source"] for document in mongo_response],
+            pit_id=None,
+            search_after=search_after,
+        )
+
+    def query_statements_by_ids(self, ids: List[str]) -> List:
```

Comment:
Maybe a silly question: why should we return a list ?

Reply:
That's a great question! For the current usage, it would be enough to return a boolean to know whether any statement matches one of the ids. However, I think, we anticipated the following requirement of the xAPI spec:

> If the LRS receives a Statement with an id it already has a Statement for, it SHOULD verify the received Statement matches the existing one and SHOULD return 409 Conflict if they do not match. See [Statement comparison requirements](https://github.com/adlnet/xAPI-Spec/blob/master/xAPI-Data.md#statement-comparison-requirements).

What return type do you suggest for this function?

Reply:
I was thinking of a more generic type, such as an Iterable

Reply:
Good Idea) :+1: Do we update this here or in a separate PR?

Reply:
Ok) I think it would be faster to add this change here) Updated it)